command/agent/command.go

```diff
@@ -641,6 +642,32 @@ func (c *Command) Run(args []string) int {
 		defer server.Shutdown()
 	}
 
+	// Enable child process reaping
+	if !config.DisableReap && (os.Getpid() == 1) {
```

Comment:
Unlike Consul Template's implementation, this implementation does not allow the user to opt-in to this behavior even if Consul is _not_ PID 1.

I think the config option should be Reap or ReapChildren and we default the value to `os.Getpid() == 1`.

Thoughts?